src/Adapter/AttributeGroup/Repository/AttributeGroupRepository.php

```diff
@@ -48,12 +52,69 @@ class AttributeGroupRepository extends AbstractMultiShopObjectModelRepository
      */
     private $dbPrefix;
 
+    /**
+     * @var AttributeGroupValidator
+     */
+    private $validator;
+
     public function __construct(
         Connection $connection,
-        string $dbPrefix
+        string $dbPrefix,
+        AttributeGroupValidator $validator
     ) {
         $this->connection = $connection;
         $this->dbPrefix = $dbPrefix;
+        $this->validator = $validator;
+    }
+
+    /**
+     * @param AttributeGroupId $attributeGroupId
+     *
+     * @return AttributeGroup
+     *
+     * @throws AttributeGroupNotFoundException
+     * @throws CoreException
+     */
+    public function get(AttributeGroupId $attributeGroupId): AttributeGroup
+    {
+        /** @var AttributeGroup $attributeGroup */
+        $attributeGroup = $this->getObjectModel(
+            $attributeGroupId->getValue(),
+            AttributeGroup::class,
+            AttributeGroupNotFoundException::class
+        );
+
+        return $attributeGroup;
+    }
+
+    /**
+     * @param AttributeGroup $attributeGroup
+     * @param int $errorCode
+     *
+     * @return AttributeGroupId
+     *
+     * @throws CoreException
+     */
+    public function add(AttributeGroup $attributeGroup, int $errorCode = 0): AttributeGroupId
```

Comment:
```suggestion
    public function add(AttributeGroup $attributeGroup): AttributeGroupId
```
I don't think you need this one on the public method since you never use it, and there's only one error case